lib/utils.js

```diff
@@ -187,24 +186,27 @@ utils.readFileGlobs = async function readFileGlobs(patterns, callback) {
  * @param {function} callback - Callback to execute for each read file.
  * @returns {Promise}
  */
-utils.readFiles = async function readFiles(files, callback) {
-  if (!files || (Array.isArray(files) && !files.length)) return Promise.resolve();
+utils.readFiles = function readFiles(files, callback) {
+  if (!files || (Array.isArray(files) && !files.length)) return Promise.resolve([]);
 
   files = (!Array.isArray(files) ? [files] : files);
-  let promises = await Promise.all(files.map(async (file) => {
+  return Promise.all(files.map(async (file) => {
     let data = await utils._readFileWithPromise(file);
     return [data, file];
-  }));
-
-  return Promise.all(promises).then(files => {
-    if (callback) {
-      files.forEach(function(data) {
-        callback.apply(null, data);
-      });
-    }
-
-    return files;
-  })
+  }))
+    .then(async files => {
+      if (callback) {
+        const promises = [];
+        files.forEach(function(data) {
+          const result = callback.apply(null, data);
+          if ('object' === typeof result && result.then) {
```

Comment:
Interesting. How are you using the callback function in this case? I'm curious what the returned Promise is doing.

Reply:
Also, wouldn't it be easier to ask if it was Promise? `if (result instanceof Promise)`